### PR TITLE
MRG: Perform Epochs rejection after ICA/SSP; add new ica_reject setting

### PR DESCRIPTION
**Pull request by hoechenberger**
Closes #300

Let's see how this goes …

This will also make it easier to plug in `autoreject` before and after ICA later!

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -1522,10 +1545,21 @@ def get_reject() -> dict:
...
+
+    return reject
+
+
+def get_reject() -> Dict[str, float]:
+    reject = _get_reject(reject=reject, ch_types=ch_types)
+    return reject
+
+
+def get_ica_reject() -> dict:
```
Comment (hoechenberger) [inline config.py:1560]:
```suggestion
def get_ica_reject() -> Dict[str, float]:
```

File: run.py [hunk 2]
```diff
@@ -28,7 +28,8 @@
     '04a-run_ica.py',
     '04b-run_ssp.py',
     '05a-apply_ica.py',
-    '05b-apply_ssp.py'
+    '05b-apply_ssp.py',
+    '06-ptp_atifacts'
```
Comment (hoechenberger) [inline run.py:32]:
```suggestion
    '06-ptp_atifacts.py'
```

File: run.py [hunk 3]
```diff
@@ -28,7 +28,8 @@
     '04a-run_ica.py',
     '04b-run_ssp.py',
     '05a-apply_ica.py',
-    '05b-apply_ssp.py'
+    '05b-apply_ssp.py',
+    '06-ptp_atifacts.py'
```
Comment (hoechenberger) [inline run.py:32]:
```suggestion
    '06-ptp_artifacts.py'
```

File: scripts/preprocessing/06-ptp_artifacts.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    fname_in = bids_path.copy().update(processing=infile_processing)
+    fname_out = bids_path.copy().update(processing='clean')
+
+    msg = f'Input: {fname_in}, Output: {fname_out}'
+    logger.info(gen_log_message(message=msg, step=6, subject=subject,
+                                session=session))
+
+    reject = config.get_reject()
+    epochs = mne.read_epochs(fname_in, preload=True)
+    epochs.drop_bad(reject=reject)
```
Comment (hoechenberger) [inline scripts/preprocessing/06-ptp_artifacts.py:54]: Deal with reject_tmin, tmax

File: run.py [hunk 5]
```diff
@@ -28,7 +28,8 @@
     '04a-run_ica.py',
     '04b-run_ssp.py',
     '05a-apply_ica.py',
-    '05b-apply_ssp.py'
+    '05b-apply_ssp.py',
+    '06-ptp_artifacts.py'
```
Comment (agramfort) [inline run.py:32]: I would rename to ptp_reject.py

Comment (hoechenberger) [reply run.py:32]: Good idea!!!

**Comment by hoechenberger** [comment]
@agramfort This is ready to merge once we go green. I'm not happy with all results for ERP CORE, but this is stuff related to bad channel marking (in the BIDS dataset) and potentially filtering, ICA, and rejection parameter adjustments. We **do** get usable grand mean results though; but there's room for improvement on per-subject analysis level. I am going to address this later on. I'd like to get this one merged first so we can move one step at a time.

File: config.py [hunk 6]
```diff
@@ -825,6 +780,26 @@
...
+The BIDS Pipeline will automatically try to detect EOG and ECG artifacts in
+your data, and remove them. For this to work properly, it is recommended
+to **not** specify rejection thresholds for EOG and ECG channels here –
+otherwise, ICA won't be able to "see" these artifacts.
+
+???+ example "Example"
+    ```python
+    reject = {'grad': 4000e-13, 'mag': 4e-12, 'eeg': 200e-6}
+    reject = {'grad': 4000e-13}
+    reject = None
```
Comment (agramfort) [inline config.py:799]: You mean ica_reject

Comment (agramfort) [reply config.py:799]: Also you should recommend much higher values than for reject

Comment (hoechenberger) [inline config.py:799]:
```suggestion
    ica_reject = {'grad': 10e-10, 'mag': 20e-12, 'eeg': 200e-6}
    ica_reject = {{'grad': 15e-10}
    ica_reject = None
```

Comment (hoechenberger) [reply config.py]: Better now?

File: config.py [hunk 7]
```diff
@@ -1522,12 +1544,23 @@ def get_reject() -> dict:
...
+            del reject[ch_type]
         except KeyError:
             pass
+
+    return reject
+
+
+def get_reject() -> Dict[str, float]:
+    reject_ = _get_reject(reject=reject, ch_types=ch_types)
     return reject_
```
Comment (agramfort) [inline config.py:1556]:
```suggestion
    return _get_reject(reject=reject, ch_types=ch_types)
```

File: config.py [hunk 8]
```diff
@@ -1522,12 +1544,23 @@ def get_reject() -> dict:
...
 
 
+def get_ica_reject() -> Dict[str, float]:
+    reject = _get_reject(reject=ica_reject, ch_types=ch_types)
+    return reject
```
Comment (agramfort) [inline config.py:1561]:
```suggestion
    return _get_reject(reject=ica_reject, ch_types=ch_types)
```

**Review by agramfort — APPROVED** [review]
just nitpicks

**Comment by agramfort** [comment]
Thx @hoechenberger

**Pull request merged**